### PR TITLE
User management

### PR DESCRIPTION
**Pull request by jeabakker**

### CODE REVIEW

File: views/default/admin/users/listing/attributes.php [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+	[
+		elgg_echo('table_columns:fromView:time_updated'),
+		elgg_view('output/datetime-local', ['value' => $entity->time_updated, 'format' => elgg_echo('friendlytime:date_format')]),
+	],
+	[
+		elgg_echo('table_columns:fromProperty:last_action'),
+		elgg_view('output/datetime-local', ['value' => $entity->last_action, 'format' => elgg_echo('friendlytime:date_format')]),
+	],
+	[
+		elgg_echo('table_columns:fromProperty:last_login'),
```
Comment (jdalsem) [inline views/default/admin/users/listing/attributes.php:39]: fromView??

File: views/default/admin/users/listing/attributes.php [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+		elgg_view('output/datetime-local', ['value' => $entity->last_login, 'format' => elgg_echo('friendlytime:date_format')]),
+	],
+	[
+		elgg_echo('table_columns:fromProperty:prev_last_login'),
```
Comment (jdalsem) [inline views/default/admin/users/listing/attributes.php:43]: fromView??

File: views/default/admin/users/listing/details.php [hunk 3]
```diff
@@ -0,0 +1,48 @@
+<?php
+/**
+ * Ajax view to list user details  on admin listings
```
Comment (jdalsem) [inline views/default/admin/users/listing/details.php:3]: double spaces

File: views/default/admin/users/listing/details.php [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+}
+
+if (!$entity instanceof \ElggUser) {
+	return;
+}
+
+$tabs = [];
+
+// user attributes
+$tabs[] = [
```
Comment (jdalsem) [inline views/default/admin/users/listing/details.php:27]: why not directly create this in the $tabs array?

Comment (jdalsem) [reply views/default/admin/users/listing/details.php:27]: and if so put it directly in the elgg_view page/components/tabs

File: views/default/admin/users/listing/details.php [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+	'text' => elgg_echo('admin:users:datails:attributes'),
+	'content' => elgg_view('admin/users/listing/attributes', $vars),
+	'selected' => true,
+];
+
+// content statistics
+$tabs[] = [
+	'text' => elgg_echo('admin:users:datails:statistics'),
```
Comment (jdalsem) [inline views/default/admin/users/listing/details.php:35]: datails everywhere!!!!

File: views/default/admin/users/listing/profile.php [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+
+$fields = elgg()->fields->get('user', 'user');
+if (empty($fields)) {
+	echo elgg_view('page/components/no_results', [
+		'no_results' => elgg_echo('admin:users:datails:profile:no_fields'),
+	]);
+	return;
+}
+
+$microformats = [
```
Comment (jdalsem) [inline views/default/admin/users/listing/profile.php:22]: do we really need this here?

File: views/default/admin/users/online.php [hunk 7]
```diff
@@ -1,3 +1,44 @@
...
+use Elgg\Database\QueryBuilder;
+use Elgg\Values;
+
+echo elgg_view('admin/users/header', [
+	'filter' => 'online',
+]);
+
+$buttons = [
+	[
+		'#type' => 'submit',
```
Comment (jdalsem) [inline views/default/admin/users/online.php:16]: same comment about code duplication

File: views/default/admin/users/unvalidated.php [hunk 8]
```diff
@@ -5,30 +5,55 @@
...
-}
+$buttons = [
+	[
+		'#type' => 'submit',
+		'icon' => 'check',
+		'value' => elgg_echo('validate'),
+		'formaction' => elgg_generate_action_url('admin/user/bulk/validate', [], false),
+	],
+	[
+		'#type' => 'submit',
```
Comment (jdalsem) [inline views/default/admin/users/unvalidated.php:20]: something about code duplication

File: views/default/core/settings/statistics/numentities.php [hunk 9]
```diff
@@ -13,25 +13,25 @@
...
 
-$rows = '';
-
-foreach ($entity_stats as $k => $entry) {
-	foreach ($entry as $a => $b) {
-		if (elgg_language_key_exists("collection:{$k}:{$a}")) {
-			$a = elgg_echo("collection:{$k}:{$a}");
-		} else {
-			$a = "$k $a";
+$rows = [];
```
Comment (jdalsem) [inline views/default/core/settings/statistics/numentities.php:16]: can't we have a helper view for entity stats (as you also use it on the admin user details)

File: views/default/forms/admin/users/bulk_actions.js [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+	
+	function disable_submit() {
+		$('form.elgg-form-admin-users-bulk-actions').find('button[type="submit"]').prop('disabled', true);
+	};
+	
+	function enable_submit() {
+		$('form.elgg-form-admin-users-bulk-actions').find('button[type="submit"]').prop('disabled', false);
+	}
+	
+	$(document).on('change', '.elgg-admin-users input[name="user_guids"]', function() {
```
Comment (jdalsem) [inline views/default/forms/admin/users/bulk_actions.js:11]: change form name to elgg-form-admin-users-bulk-actions

File: views/default/forms/admin/users/bulk_actions.php [hunk 11]
```diff
@@ -0,0 +1,82 @@
...
+$options = (array) elgg_extract('options', $vars, []);
+$options = array_merge($default_options, $options);
+
+$users = elgg_call(ELGG_SHOW_DISABLED_ENTITIES, function() use ($options, $getter) {
+	return elgg_list_entities($options, $getter);
+});
+if (empty($users)) {
+	$no_results = elgg_extract('no_results', $vars);
+	if (empty($no_results) || $no_results === true) {
+		$no_results = elgg_echo('notfound');
```
Comment (jdalsem) [inline views/default/forms/admin/users/bulk_actions.php:60]: why not add the === true behaviour to page/components/no_results??

File: views/default/forms/admin/users/bulk_actions.php [hunk 12]
```diff
@@ -0,0 +1,82 @@
...
+if (!empty($buttons)) {
+	foreach ($buttons as &$button) {
+		$button['disabled'] = true;
+	}
+	
+	echo elgg_view_field([
+		'#type' => 'fieldset',
+		'#class' => ['elgg-admin-users-bulkactions-buttons', 'mbs'],
+		'align' => 'horizontal',
+// 		'justify' => 'right',
```
Comment (jdalsem) [inline views/default/forms/admin/users/bulk_actions.php:77]: huh

File: views/default/page/components/column/entity_menu.php [hunk 13]
```diff
@@ -0,0 +1,19 @@
...
+
+$item = elgg_extract('item', $vars);
+if (!$item instanceof \ElggEntity) {
+	return;
+}
+
+echo elgg_view_menu('entity', [
+	'entity' => $item,
+	'prepare_dropdown' => true,
+	'add_user_hover_admin_section' => (bool) elgg_extract('add_user_hover_admin_section', $vars, false),
```
Comment (jdalsem) [inline views/default/page/components/column/entity_menu.php:18]: maybe only allow this flag if $item instanceof \ElggUser???

File: views/default/page/components/column/icon.php [hunk 14]
```diff
@@ -14,5 +14,7 @@
 }
 
 $size = elgg_extract('size', $vars, 'small');
+unset($vars['item']);
+unset($vars['size']);
 
-echo elgg_view_entity_icon($item, $size);
+echo elgg_view_entity_icon($item, $size, $vars);
```
Comment (jdalsem) [inline views/default/page/components/column/icon.php:20]: could you check if this leaks unwanted stuff into html?

File: views/default/page/components/column/select.php [hunk 15]
```diff
@@ -0,0 +1,32 @@
+<?php
```
Comment (jdalsem) [inline views/default/page/components/column/select.php:1]: rename file??

File: views/default/page/components/column/unvalidated_menu.php [hunk 16]
```diff
@@ -0,0 +1,19 @@
+<?php
+/**
+ * Render an entity menu element
```
Comment (jdalsem) [inline views/default/page/components/column/unvalidated_menu.php:3]: unvalidated menu... copy paste :)

**Review by jdalsem — CHANGES_REQUESTED** [review]

File: views/default/forms/admin/users/bulk_actions.php (hunk 11 above)
Comment (jeabakker) [reply views/default/forms/admin/users/bulk_actions.php:60]: maybe later

File: views/default/page/components/column/entity_menu.php (hunk 13 above)
Comment (jeabakker) [reply views/default/page/components/column/entity_menu.php:18]: leave as as, because handling only happens for users

**Review by jdalsem — APPROVED** [review]

**Pull request merged**